Expanded property and Msv class names to human-friendly names

tidoust opened the pull request:
It is usually considered good practice to define human-friendly class and
property names in specification and to avoid acronyms when possible.

This update changes "Msv" used as a concept into "media state vector" and
"Msv" used as a class name into "MediaStateVector". Media state vector
properties have also been renamed to "position", "velocity", "acceleration"
and "timestamp" to improve readability.

I also removed commas next to "i.e." in that commit. While I understand
that it's actually correct in American English, all specs I checked do not
use it, and it catches my eye each time. I hid that removal into this commit
in the hope that the change would go unnoticed ;)

tidoust committed:
Expanded property and Msv class names to human-friendly names

It is usually considered good practice to define human-friendly class and
property names in specification and to avoid acronyms when possible.

This update changes "Msv" used as a concept into "media state vector" and
"Msv" used as a class name into "MediaStateVector". Media state vector
properties have also been renamed to "position", "velocity", "acceleration"
and "timestamp" to improve readability.

I also removed commas next to "i.e." in that commit. While I understand
that it's actually correct in American English, all specs I checked do not
use it, and it catches my eye each time. I hid that removal into this commit
in the hope that the change would go unnoticed ;)

tidoust committed:
Expanded property and Msv class names to human-friendly names

It is usually considered good practice to define human-friendly class and
property names in specification and to avoid acronyms when possible.

This update changes "Msv" used as a concept into "media state vector" and
"Msv" used as a class name into "MediaStateVector". Media state vector
properties have also been renamed to "position", "velocity", "acceleration"
and "timestamp" to improve readability.

I also removed commas next to "i.e." in that commit. While I understand
that it's actually correct in American English, all specs I checked do not
use it, and it catches my eye each time. I hid that removal into this commit
in the hope that the change would go unnoticed ;)

ingararntzen commented:
How about naming is "TimingStateVector" or even better simply "StateVector"? "MediaStateVector" was originally used as a concept name - and as such it has been replaced by "Shared Motion". In this context of internal programming structures there might be no need to keep the prefix "media" at all as it possibly only serves to add confusion or limit application scope?

ingararntzen committed:
Merge pull request #4 from tidoust/fix-eventhandlers

Fixed the EventHandler definitions

tidoust commented:
That's a good point. My main concern here was about avoiding acronyms, but it's true that "media" is indeed confusing.

"StateVector" makes me think of something standing still, which is not exactly the image I would like to have for something that is to represent motion, but "MotionVector" seems to be used in MPEG video processing so probably not a good idea to use that term to mean something else. So let's go for "StateVector".

Should I update my pull request accordingly?

ingararntzen commented:
Actually, I think the "standing still" interpretation may even be helpful.
The state vector is in a sense a static serialization of a live motion,
allowing it to be exchanged over a network and de-serialized by other
devices.

Yes, please update the pull request :)

Ingar

2015-06-08 10:45 GMT+02:00 François Daoust notifications@github.com:

> That's a good point. My main concern here was about avoiding acronyms, but
> it's true that "media" is indeed confusing.
> 
> "StateVector" makes me think of something standing still, which is not
> exactly the image I would like to have for something that is to represent
> motion, but "MotionVector" seems to be used in MPEG video processing so
> probably not a good idea to use that term to mean something else. So let's
> go for "StateVector".
> 
> Should I update my pull request accordingly?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/webtiming/timingobject/pull/5#issuecomment-109910668.

tidoust committed:
Renamed "media state vector" into "state vector" based on discussion

I also extended the scope of the "vector" definition to include "internal"
since the spec always talks about internal vectors and not about vectors
as such.

tidoust committed:
Renamed "media state vector" into "state vector" based on discussion

I also extended the scope of the "vector" definition to include "internal"
since the spec always talks about internal vectors and not about vectors
as such.

tidoust committed:
Merge branch 'expandednames' of github.com:tidoust/timingobject into expandednames

Conflicts:
	index.html

tidoust committed:
Renamed "MediaStateVector" into "StateVector" as well

tidoust commented:
Done. Not sure why some commit appear twice in the pull request but the final result should be good.
